src/Language/PureScript/Sugar/BindingGroups.hs

```diff
@@ -137,18 +159,21 @@ flattenBindingGroups = concatMap go
       ValueDecl sa ident nameKind [] [MkUnguarded val]) ds
   go other = [other]
 
-usedIdents :: ModuleName -> ValueDeclarationData Expr -> [Ident]
-usedIdents moduleName = ordNub . usedIdents' S.empty . valdeclExpression
+usedIdents :: M.Map Ident Bool -> ModuleName -> ValueDeclarationData Expr -> [(Bool, Ident)]
+usedIdents identInfo moduleName = ordNub . usedIdents' S.empty . valdeclExpression
```

Comment:
This change strikes me as an awkward mixing of responsibilities; any reason not to map over the result up in `handleDecls` and leave this unchanged?

Reply:
Good catch! Reverting this...